examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2218 @@
+                                   indices_relevant.end(),
+                                   on_subranges,
+                                   4096);
+    } /* End of the computation of the off-diagonal entries of dij_matrix */
+
+    std::atomic<double> tau_max{std::numeric_limits<double>::infinity()};
+
+    {
+      TimerOutput::Scope time(computing_timer,
+                              "time_step - 2 compute d_ii, and tau_max");
+
+      const auto on_subranges = [&](auto i1, const auto i2) {
+        double tau_max_on_subrange = std::numeric_limits<double>::infinity();
+
+        for (const auto i : boost::make_iterator_range(i1, i2))
+          {
+            double d_sum = 0.;
+
+            for (auto jt = sparsity.begin(i); jt != sparsity.end(i); ++jt)
+              {
+                const auto j = jt->column();
+
+                if (j == i)
+                  continue;
+
+                d_sum -= get_entry(dij_matrix, jt);
+              }
+
+            dij_matrix.diag_element(i) = d_sum;
+
+            const double mass   = lumped_mass_matrix.diag_element(i);
+            const double tau    = cfl_update * mass / (-2. * d_sum);
+            tau_max_on_subrange = std::min(tau_max_on_subrange, tau);
+          }
+
+        double current_tau_max = tau_max.load();
+        while (
+          current_tau_max > tau_max_on_subrange &&
+          !tau_max.compare_exchange_weak(current_tau_max, tau_max_on_subrange))
+          ;
+      };
+
+      parallel::apply_to_subranges(indices_relevant.begin(),
+                                   indices_relevant.end(),
+                                   on_subranges,
+                                   4096);
+
+      tau_max.store(Utilities::MPI::min(tau_max.load(), mpi_communicator));
+
+      AssertThrow(!std::isnan(tau_max) && !std::isinf(tau_max) && tau_max > 0.,
+                  ExcMessage("I'm sorry, Dave. I'm afraid I can't "
+                             "do that. - We crashed."));
+    } /* End of the computation of the diagonal entries of dij_matrix */
+
+    {
+      TimerOutput::Scope time(computing_timer, "time_step - 3 perform update");
+
+      const auto on_subranges = [&](auto i1, const auto i2) {
+        for (const auto i : boost::make_iterator_range(i1, i2))
+          {
+            Assert(i < n_locally_owned, ExcInternalError());
+
+            const auto U_i = gather(U, i);
+
+            const auto   f_i = ProblemDescription<dim>::f(U_i);
+            const double m_i = lumped_mass_matrix.diag_element(i);
+
+            auto U_i_new = U_i;
+
+            for (auto jt = sparsity.begin(i); jt != sparsity.end(i); ++jt)
+              {
+                const auto j = jt->column();
+
+                const auto U_j = gather(U, j);
+                const auto f_j = ProblemDescription<dim>::f(U_j);
+
+                const auto c_ij = gather_get_entry(cij_matrix, jt);
+                const auto d_ij = get_entry(dij_matrix
```

Comment:
you should use the new write_vtu_with_pvtu_record

Reply:
(again, looks like github placed the comment in the wrong place. This was about the write_vtk())